frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditMobile/EditMobile.tsx

```diff
@@ -0,0 +1,131 @@
+
+import { useFreeSmsQuota } from '~features/admin-form/common/queries'
+
+import { useCreateTabForm } from '../../../../useCreateTabForm'
+import { DrawerContentContainer } from '../common/DrawerContentContainer'
+import { FormFieldDrawerActions } from '../common/FormFieldDrawerActions'
+import { EditFieldProps } from '../common/types'
+import { useEditFieldForm } from '../common/useEditFieldForm'
+
+import { SmsCountMessage } from './SmsCountMessage'
+import { SmsCountsModal } from './SmsCountsModal'
+import { TwilioCredentialsMessage } from './TwilioCredentialsMessage'
+
+const EDIT_MOBILE_KEYS = [
+  'title',
+  'description',
+  'required',
+  'isVerifiable',
+  'allowIntlNumbers',
+] as const
+
+type EditMobileProps = EditFieldProps<MobileFieldBase>
+
+type EditMobileInputs = Pick<MobileFieldBase, typeof EDIT_MOBILE_KEYS[number]>
+
+export const EditMobile = ({ field }: EditMobileProps): JSX.Element => {
+  const {
+    register,
+    formState: { errors },
+    isSaveEnabled,
+    buttonText,
+    handleUpdateField,
+    isLoading,
+    handleCancel,
+  } = useEditFieldForm<EditMobileInputs, MobileFieldBase>({
+    field,
+    transform: {
+      input: (inputField) => pick(inputField, EDIT_MOBILE_KEYS),
+      output: (formOutput, originalField) =>
+        extend({}, originalField, formOutput),
+    },
+  })
+
+  const requiredValidationRule = useMemo(
+    () => createBaseValidationRules({ required: true }),
+    [],
+  )
+
+  const { data: form } = useCreateTabForm()
+  const hasTwilioCredentials = useMemo(() => !!form?.msgSrvcName, [form])
+
+  const { data: freeSmsCount } = useFreeSmsQuota()
+  const smsCountsModal = useDisclosure()
+
+  return (
+    <>
+      <DrawerContentContainer>
+        <FormControl
+          isRequired
+          isReadOnly={isLoading}
+          isInvalid={!!errors.title}
+        >
+          <FormLabel>Question</FormLabel>
+          <Input autoFocus {...register('title', requiredValidationRule)} />
+          <FormErrorMessage>{errors?.title?.message}</FormErrorMessage>
+        </FormControl>
+        <FormControl
+          isRequired
+          isReadOnly={isLoading}
+          isInvalid={!!errors.description}
+        >
+          <FormLabel>Description</FormLabel>
+          <Textarea {...register('description')} />
+          <FormErrorMessage>{errors?.description?.message}</FormErrorMessage>
+        </FormControl>
+        <FormControl isReadOnly={isLoading}>
+          <Toggle {...register('required')} label="Required" />
+        </FormControl>
+        <FormControl isReadOnly={isLoading}>
+          <Toggle
+            {...register('allowIntlNumbers')}
+            label="Allow international numbers"
+          />
+        </FormControl>
+        <Box>
+          <FormControl isReadOnly={isLoading}>
+            <Toggle
+              {...register('isVerifiable', {
+                onChange: (e) => {
+                  if (e.target.checked) {
+                    smsCountsModal.onOpen()
```

Comment:
do we want to open the sms counts modal whenever its toggled? What about when the form already has twilio credentials?